### PR TITLE
Reorganize .gitignore

### PR DESCRIPTION
**Pull request by Pastafarianist**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,24 +1,8 @@
...
-*.dyn_hi
-*.prof
-*.aux
-*.hp
-*.tix
 .virtualenv
 .hsenv
 .hpc
-.Werror
 *.svg
```
Comment (volhovM) [inline .gitignore]: Should be removed together with ipython-related things.

File: .gitignore [hunk 2]
```diff
@@ -27,11 +11,7 @@ TAGS
 *~
```
Comment (volhovM) [inline .gitignore]: Section: emacs files.

File: .gitignore [hunk 3]
```diff
@@ -10,68 +9,63 @@ cabal-dev
...
+
+# Cabal
+cabal-dev
 .cabal-sandbox/
 cabal.sandbox.config
 cabal.config
-GTAGS
-GRTAGS
-GPATH
-TAGS
```
Comment (gromakovsky) [inline .gitignore]: Why did you delete e. g. `TAGS`? IIRC there is some software which dumps this file and perhaps some people use it.

File: .gitignore [hunk 4]
```diff
@@ -10,68 +9,63 @@ cabal-dev
...
 wallet-db/
-zhogovo/
-smartgen-secret.sk
-rocks-smartwallet/
-cardano-keys/
-bla/
-testnetStakes/
-*.key.lock
-node_modules/*
 db-abc/
```
Comment (gromakovsky) [inline .gitignore]: It's obsolete I suppose. Grep finds `abc` in scripts only in `scripts/avvm-files/utxo-dump-last-new.json` and `scripts/clean/db.sh`.
